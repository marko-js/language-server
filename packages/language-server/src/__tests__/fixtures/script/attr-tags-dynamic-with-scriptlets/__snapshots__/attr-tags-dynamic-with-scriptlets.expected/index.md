## Hovers
### Ln 5, Col 11
```marko
  3 |
  4 | <${custom}>
> 5 |   <while(!done)>
    |           ^ let done: false
  6 | //        ^?
  7 |     $ i++;
  8 | //    ^?
```

### Ln 7, Col 7
```marko
   5 |   <while(!done)>
   6 | //        ^?
>  7 |     $ i++;
     |       ^ let i: number
   8 | //    ^?
   9 |     <@a>
  10 |       ${done}
```

### Ln 10, Col 9
```marko
   8 | //    ^?
   9 |     <@a>
> 10 |       ${done}
     |         ^ let done: false
  11 | //      ^?
  12 |       <if(i === 5)>
  13 |         $ done = true;
```

### Ln 21, Col 13
```marko
  19 | <${custom}>
  20 |   <for|index| from=1 to=10>
> 21 |     $ const doubleIndex = index * 2;
     |             ^ const doubleIndex: number
  22 | //          ^?
  23 |     <@a>
  24 |       ${doubleIndex}
```

### Ln 24, Col 9
```marko
  22 | //          ^?
  23 |     <@a>
> 24 |       ${doubleIndex}
     |         ^ const doubleIndex: number
  25 | //      ^?
  26 |     </@a>
  27 |   </for>
```

### Ln 32, Col 13
```marko
  30 | <${custom} x=1>
  31 |   <if=x>
> 32 |     $ const a = 1 as const;
     |             ^ const a: 1
  33 | //          ^?
  34 |     <@a a=a/>
  35 |   </if>
```

### Ln 37, Col 13
```marko
  35 |   </if>
  36 |   <else>
> 37 |     $ const b = 2 as const;
     |             ^ const b: 2
  38 | //          ^?
  39 |     <@b b=b/>
  40 |   </else>
```

## Diagnostics
### Ln 4, Col 4
```marko
  2 | $ let i = 0;
  3 |
> 4 | <${custom}>
    |    ^^^^^^ Cannot find name 'custom'.
  5 |   <while(!done)>
  6 | //        ^?
  7 |     $ i++;
```

### Ln 19, Col 4
```marko
  17 | </>
  18 |
> 19 | <${custom}>
     |    ^^^^^^ Cannot find name 'custom'.
  20 |   <for|index| from=1 to=10>
  21 |     $ const doubleIndex = index * 2;
  22 | //          ^?
```

### Ln 30, Col 4
```marko
  28 | </>
  29 |
> 30 | <${custom} x=1>
     |    ^^^^^^ Cannot find name 'custom'.
  31 |   <if=x>
  32 |     $ const a = 1 as const;
  33 | //          ^?
```

### Ln 31, Col 7
```marko
  29 |
  30 | <${custom} x=1>
> 31 |   <if=x>
     |       ^ Cannot find name 'x'.
  32 |     $ const a = 1 as const;
  33 | //          ^?
  34 |     <@a a=a/>
```

