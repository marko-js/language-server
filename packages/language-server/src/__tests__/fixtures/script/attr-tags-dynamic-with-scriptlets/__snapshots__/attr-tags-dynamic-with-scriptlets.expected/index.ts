export interface Input {}
abstract class Component extends Marko.Component<Input> {}
export { type Component };
(function (this: void) {
  const input = Marko._.any as Input;
  const component = Marko._.any as Component;
  const state = Marko._.state(component);
  const out = Marko._.out;
  const $signal = Marko._.any as AbortSignal;
  const $global = Marko._.getGlobal(
    // @ts-expect-error We expect the compiler to error because we are checking if the MarkoRun.Context is defined.
    (Marko._.error, Marko._.any as MarkoRun.Context),
  );
  Marko._.noop({ component, state, out, input, $global, $signal });
  let done = false;
  let i = 0;
  const __marko_internal_tag_1 = custom;
  Marko._.attrTagNames(__marko_internal_tag_1, (input) => {
    input["@a"];
  });
  Marko._.renderDynamicTag(__marko_internal_tag_1)()()({
    ...(!done
      ? (() => {
          //        ^?
          i++;
          return {
            //    ^?
            ["a" /*@a*/]: {
              ["renderBody" /*@a*/]: (() => {
                done;
                //      ^?
                if (i === 5) {
                  done = true;
                }
                return () => {
                  return Marko._.voidReturn;
                };
              })(),
              [/*@a*/ Symbol.iterator]: Marko._.any,
            },
          };
        })()
      : {}),
  });
  const __marko_internal_tag_2 = custom;
  Marko._.attrTagNames(__marko_internal_tag_2, (input) => {
    input["@a"];
  });
  Marko._.renderDynamicTag(__marko_internal_tag_2)()()({
    ...Marko._.forToAttrTag(
      {
        /*for*/ from: 1,
        to: 10,
      },
      (index) =>
        (() => {
          const doubleIndex = index * 2;
          return {
            //          ^?
            ["a" /*@a*/]: {
              ["renderBody" /*@a*/]: (() => {
                doubleIndex;
                return () => {
                  return Marko._.voidReturn;
                };
              })(),
              [/*@a*/ Symbol.iterator]: Marko._.any,
            },
          };
        })(),
    ),
  });
  const __marko_internal_tag_3 = custom;
  Marko._.attrTagNames(__marko_internal_tag_3, (input) => {
    input["@a"];
    input["@b"];
  });
  Marko._.renderDynamicTag(__marko_internal_tag_3)()()(
    //      ^?
    {
      x: 1,
      ...(x
        ? (() => {
            const a = 1 as const;
            return {
              //          ^?
              ["a" /*@a*/]: {
                a: a,
                [/*@a*/ Symbol.iterator]: Marko._.any,
              },
            };
          })()
        : (() => {
            const b = 2 as const;
            return {
              //          ^?
              ["b" /*@b*/]: {
                b: b,
                [/*@b*/ Symbol.iterator]: Marko._.any,
              },
            };
          })()),
    },
  );
  return;
})();
export default new (class Template extends Marko._.Template<{
  render(
    input: Marko.TemplateInput<Input>,
    stream?: {
      write: (chunk: string) => void;
      end: (chunk?: string) => void;
    },
  ): Marko.Out<Component>;

  render(
    input: Marko.TemplateInput<Input>,
    cb?: (err: Error | null, result: Marko.RenderResult<Component>) => void,
  ): Marko.Out<Component>;

  renderSync(input: Marko.TemplateInput<Input>): Marko.RenderResult<Component>;

  renderToString(input: Marko.TemplateInput<Input>): string;

  stream(
    input: Marko.TemplateInput<Input>,
  ): ReadableStream<string> & NodeJS.ReadableStream;

  api: "class";
  _(): () => <__marko_internal_input extends unknown>(
    input: Marko.Directives &
      Input &
      Marko._.Relate<__marko_internal_input, Marko.Directives & Input>,
  ) => Marko._.ReturnWithScope<__marko_internal_input, void>;
}> {})();
