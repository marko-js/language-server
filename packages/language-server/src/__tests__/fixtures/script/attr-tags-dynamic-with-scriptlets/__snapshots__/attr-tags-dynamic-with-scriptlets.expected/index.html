<div>
  <div>

    

    
      placeholder

      <div>
        
      </div>
    
  </div>
</div><div>
  <div>
    

    
      placeholder

    
  </div>
</div><div>
  <div>
    

    
  </div>
  <div>
    

    
  </div>
</div>